src/main/java/com/achdev/onlinebookstoreapp/repository/user/UserRepository.java

```diff
@@ -1,8 +1,13 @@
 package com.achdev.onlinebookstoreapp.repository.user;
 
 import com.achdev.onlinebookstoreapp.model.User;
+import java.util.Optional;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
 
 public interface UserRepository extends JpaRepository<User, Long> {
     boolean existsByEmail(String email);
+
+    @Query(value = "SELECT u FROM User u LEFT JOIN FETCH u.roles WHERE u.email = :email")
```

Comment:
Can users without roles exist? Let’s  use the correct type of join

Reply:
Ok